### PR TITLE
bsp: New BSP for M5Dial (BSP-488)

### PR DESCRIPTION
**Pull request by espzav**
# Checklist for new Board Support package or Component

- [x] Component contains License
- [x] Component contains README.md
- [x] Project [README.md](../README.md) updated
- [x] Component contains idf_component.yml file with `url` field defined
- [x] Component was added to CI [upload job](https://github.com/espressif/esp-bsp/blob/master/.github/workflows/upload_component.yml#L17)
- [x] New files were added to CI build job
- [x] New BSP definitions added to [bsp_ext.py](../examples/bsp_ext.py)
- [x] BSP was [added to SquareLine](https://github.com/espressif/esp-bsp/tree/master/SquareLine/common)
- [x] New BSP supports IDF v4.4.x and later releases
- [x] CI passing
- [x] Wait and update after merge #322 
- [x] Wait and update after merge #327  

# Change description

### CODE REVIEW

**Comment by espzav** [comment]
@igrr Here is PR for M5Dial. The test issues was fixed in #327 (waiting for merge)

**Comment by espzav** [comment]
@tore-espressif and @pborcin PTAL

File: bsp/m5dial/include/bsp/m5dial.h [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+#include "esp_lvgl_port.h"
+#endif // BSP_CONFIG_NO_GRAPHIC_LIB == 0
+
+/**************************************************************************************************
+ *  BSP Capabilities
+ **************************************************************************************************/
+
+#define BSP_CAPS_DISPLAY        1
+#define BSP_CAPS_TOUCH          1
+#define BSP_CAPS_BUTTONS        0
```
Comment (tore-espressif) [inline bsp/m5dial/include/bsp/m5dial.h:31]: We use button and knob components here

```suggestion
#define BSP_CAPS_BUTTONS        1
#define BSP_CAPS_KNOB              1
```

File: bsp/m5dial/include/bsp/m5dial.h [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+#define BSP_CAPS_AUDIO          0
+#define BSP_CAPS_AUDIO_SPEAKER  0
+#define BSP_CAPS_AUDIO_MIC      0
+#define BSP_CAPS_SDCARD         0
+#define BSP_CAPS_IMU            0
+
+/**************************************************************************************************
+ *  M5Stack-Core-S3 pinout
```
Comment (tore-espressif) [inline bsp/m5dial/include/bsp/m5dial.h:39]:
```suggestion
 *  M5Stack-Dial pinout
```

File: bsp/m5dial/include/bsp/m5dial.h [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+extern "C" {
+#endif
+
+/**************************************************************************************************
+ *
+ * I2C interface
+ *
+ * There are multiple devices connected to I2C peripheral:
+ *  - Codec AW88298 (configuration only)
+ *  - ADC ES7210 (configuration only)
```
Comment (tore-espressif) [inline bsp/m5dial/include/bsp/m5dial.h:74]: Outdated docs?

File: bsp/m5dial/include/bsp/m5dial.h [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+ *      - ESP_FAIL if partition can not be mounted
+ *      - other error codes
+ */
+esp_err_t bsp_spiffs_unmount(void);
+
+/**************************************************************************************************
+ *
+ * LCD interface
+ *
+ * M5Stack-Core-S3 is shipped with 2.0inch ILI9341C display controller.
```
Comment (tore-espressif) [inline bsp/m5dial/include/bsp/m5dial.h:142]: Outdated docs

**Review by tore-espressif — APPROVED** [review]
@espzav Thanks for the PR!

I tested the examples - all OK, and add few nitpick comments.

Next we can discuss support for the other board features, such as the RTC timer or RFID

**Pull request merged**